lib/mix/lib/mix/tasks/compile.app.ex

```diff
@@ -3,6 +3,23 @@ defmodule Mix.Tasks.Compile.App do
 
   @recursive true
 
+  # See http://erlang.org/doc/man/app.html.
+  @keys_whitelist [
```

Comment:
What do you think about renaming `@keys_whitelist` to `@application_specification_keys` or `@application_specification_properties`?

Reply:
Hey @fertapric :).

Regarding "keys" vs "properties", I saw documentation refers to the first element of the tuple as the key. Existing code seems to refer to "properties" as the list of tuples (3rd element of the main tuple). The test suite also uses "properties" in that sense. With those conventions, this would be a whitelist of keys.

The names you suggest would be more precise, but also more verbose. The purpose of a name like that would be for the attribute to be self-documenting, but the purpose of this task is so specialized, and the collection of atoms so recognizable visually, that I opted for a shorter name.

That said, can be changed of course (and it is only used in one place, so anything would work length-wise).